### PR TITLE
CLicodes

### PR DESCRIPTION
**Pull request by Anu2001-prog**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/mysql/custom.py [hunk 1]
```diff
@@ -57,6 +57,65 @@ def server_list_custom_func(client, resource_group_name=None):
...
+
+    if not server_name:
+        raise ValueError("Invalid server name provided.")
+
+
+    return client.list(resource_group_name,server_name)
+
+def flexible_server_threat_model_show_func(cmd, client, resource_group_name=None,
+                                            server_name=None,
+                                            advanced_threat_protection_name="default"
```
Comment (susantabiswas) [inline src/azure-cli/azure/cli/command_modules/mysql/custom.py:105]: Add a null check for it incase the user passes a value

File: src/azure-cli/azure/cli/command_modules/mysql/_params.py [hunk 2]
```diff
@@ -299,6 +299,20 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements, too-many-
         c.argument('standby_availability_zone', arg_type=standby_availability_zone_arg_type)
         c.argument('database_name', arg_type=database_name_arg_type)
         c.argument('yes', arg_type=yes_arg_type)
+    
+    with self.argument_context('mysql flexible-server threat-protection update') as c:
```
Comment (nakulms) [inline src/azure-cli/azure/cli/command_modules/mysql/_params.py:303]: Can you confirm if subscription id is required or not. Maybe it is not required as az login might take care of it. Or is it a better practice to include it as an optional parameter. Please confirm.

Comment (Anu2001-prog) [reply src/azure-cli/azure/cli/command_modules/mysql/_params.py:303]: Yes, It is handled by az login and supplied as context.

**Pull request closed without merging**